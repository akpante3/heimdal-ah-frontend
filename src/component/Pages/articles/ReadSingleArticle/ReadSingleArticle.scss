@import "../../../../styles/variables.scss";
@import "../../../../styles/global.scss";

.single-article-wrapper {
  background-color: whitesmoke;
  .article-header {
    text-align: center;
    background: #6a7390;
    font-family: "Roboto", serif;
    padding-right: 10%;
    padding-left: 10%;
    height: 200px;
    color: white;

    h1 {
      margin-top: 30px;
      padding-top: 5%;
      font-size: 40px;
    }
  }

  .bookmark-row {
    display: grid;
    grid-template-columns: 73% 27%;
    grid-gap: 1em;
    height: 80px;

    .bookmark-logo {
      color: #6a7390;
      display: "none";
      cursor: pointer;
    }
  }

  .body-section {
    display: grid;
    grid-template-columns: 75% 20%;
    text-align: justify;
    grid-gap: 1.5em;
    padding: 0 5%;
    font-family: "Roboto thin", serif;
    word-break: break-all;

    .article-section {
      background-color: whitesmoke;
      white-space: normal;
      padding: 0px 20px;
      font-size: 18px;
      color: #3e4653;
      line-height: 2;
      .body-header {
        display: grid;
        grid-template-columns: 95% 5%;
        .bookmark-logo {
          color: #6a7390;
          display: "none";
          cursor: pointer;
          font-size: 40px;
        }
        .bookmark-logo:hover {
          color: rgb(126, 125, 66);
        }
      }
      .bottom-details {
        display: grid;
        grid-template-columns: 30% 40% 30%;
        margin-top: 10%;
        .ratings {
          padding-left: 15px;
        }
        .socails {
          grid-template-columns: 30% 30% 30%;
          .span {
            display: inline-block;
            padding-right: 10px;
          }
          .facebook {
            color: #3b5998;
            outline: none;
          }
          .twitter {
            color: #1da1f2;
            outline: none;
          }
          .no-outline {
            outline: none;
            display: inline-block;
            padding-right: 20px;
          }
        }
        .likesCount {
          margin-bottom: 2px;
        }
      }
    }

    .user-section {
      background-color: whitesmoke;
      border: none;
      border-radius: 10px;
      padding: 5%;
      display: grid;
      grid-template-rows: 35% 25% 30%;
      height: 300px;
      box-shadow: 0px 1px 2px 0 rgba(0, 0, 0, 0.2);

      .follow-btn {
        background-color: #153771;
        color: white;
        align-self: center;
      }
      .user-link {
        text-decoration: none;
        color: #6a7384;
      }

      .user-profile {
        text-align: center;
        color: #6a7384;
      }

      .user-image {
        width: 100px;
        height: 100px;
        border-radius: 50%;
        margin-top: 10px;
      }

      .social-media {
        display: grid;
        grid-template-columns: 50% 50%;
        // margin-bottom: 60px;
      }

      // .no-outline {
      //   outline: none;
      // }
    }
    .glow {
      text-align: end;
    }
    .ratings {
      text-align: center;
    }
    .line {
      height: 0.8px;
      background-color: #6a7384;
    }
  }

  .tag-buttons {
    padding: 10px;
    margin-top: 3%;
    padding-left: 20px;
  }
}
.ReadSingleArticleError {
  height: 600px;
}
