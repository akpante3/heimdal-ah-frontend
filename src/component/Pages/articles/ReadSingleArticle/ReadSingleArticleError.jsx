import React from 'react';

const ReadSingleArticleError = () => (
  <div className='container ReadSingleArticleError'>
    <h3>
        Sorry This Article
      <br />
      oops something went wrong!
    </h3>
    <br />
    <a to='/'>
      <div>..go back Home....</div>
    </a>
  </div>
);

export default ReadSingleArticleError;
