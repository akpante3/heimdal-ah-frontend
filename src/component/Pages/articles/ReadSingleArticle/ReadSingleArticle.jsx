/* eslint-disable max-len */
import React, { Component, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import './ReadSingleArticle.scss';
// import setToken from '../../../../config/setToken';
import getArticleById from '../../../../actions/ArticleActions/getArticlesByIdActions';
import decodeToken from '../../../../utils/decodeToken';
import ReadSingleArticlePresentation from './ReadSingleArticlePresentation';
import getAllBookmarksAction from '../../../../actions/ArticleActions/bookmarksAction/getAllBookmarksAction';
import createBookmarkAction from '../../../../actions/ArticleActions/bookmarksAction/createBookmarkAction';
import deleteBookmarksActions from '../../../../actions/ArticleActions/bookmarksAction/deleteBookmarksActions';


/**
 * @description - Read a single article posted by a user
 * @param {props} singleArticle - the status state object that contains all details about the article
 */
export class ReadSingleArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      singleArticle: {
        title: '',
        id: '',
        userId: '',
        slug: '',
        body: '',
        likes: [],
        tags: [],
        user: {},
      },
      bookmarks: false,
    };
  }


  /**
 * @description - component mounts method runs as soon as the page loads
 * * @description token - decode gets the user token from local storage
 * * @description  - decode gets the user token from local storage
 */
  componentDidMount() {
    const { slug } = this.props.match.params;
    this.props.actions.getArticleById(slug);
    this.props.actions.getAllBookmarksAction();
    // const bookmarks = this.props.bookmark.payload.rows;

    // const isBookmarked = bookmarks && bookmarks.some(item => item.article.id === this.state.singleArticle.id);

    // if (isBookmarked) {
    //   this.setState({ bookmarks: true });
    // }
  }

  /**
 * @description - Handles props being received
 * @param {string} nextProps - props being passed
 * @returns {component} update state
 */
  componentWillReceiveProps(nextProps) {
    this.setState({ ...nextProps, singleArticle: nextProps.singleArticle });
    const bookmarks = nextProps.bookmark.payload.rows;

    const isBookmarked = bookmarks && bookmarks.some(item => item.article.id === this.state.singleArticle.id);

    if (isBookmarked) {
      this.setState({ bookmarks: true });
    }
  }

  createBookmark = () => {
    this.setState({ bookmarks: true });
    this.props.actions.createBookmarkAction(this.state.singleArticle.id);
  }

  deleteBookmark = () => {
    this.setState({ bookmarks: false });
    this.props.actions.deleteBookmarksActions(this.state.singleArticle.id);
  }

  /**
 * @description - render single article page
 * @param {props} status - the status returned from dispatching actions
 * @returns {component} the render component
 */
  render() {
    const author = decodeToken(this.state.singleArticle.userId);
    const likesCount = this.state.singleArticle.likes && this.state.singleArticle.likes.length;
    const { user = {} } = this.state.singleArticle;
    const { profile = {} } = user;
    const { slug } = this.props.match.params;
    const { status } = this.props;
    
    // const bookmarks = this.props.bookmark.payload.rows;

    // const isBookmarked = bookmarks && bookmarks.some(item => item.article.id === this.state.singleArticle.id);

    // // if (isBookmarked) {
    // //   this.setState({ bookmarks: true });
    // // }

    return (
      <Fragment>
        { status === 'ERROR' ? <Redirect to={`/articles/${slug}`} />
          : (
            <Fragment>
              <ReadSingleArticlePresentation
                createBookmark={this.createBookmark}
                deleteBookmark={this.deleteBookmark}
                bookmark={this.state.bookmarks}
                slug={slug}
                author={author}
                articleId={this.state.singleArticle.id}
                title={this.state.singleArticle.title}
                body={this.state.singleArticle.body}
                username={profile.username}
                time={this.state.singleArticle.createdAt}
                likesCount={likesCount}
                userImage={profile.image}
              />
            </Fragment>
          )
        }
      </Fragment>
    );
  }
}

ReadSingleArticlePresentation.defaultProps = {
  username: '@username',
};


ReadSingleArticlePresentation.propTypes = {
  username: PropTypes.string,
};

const mapStateToprops = state => ({
  singleArticle: state.getArticlesById.payload,
  status: state.getArticlesById.status,
  bookmark: state.getallbookmarks,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      getArticleById,
      getAllBookmarksAction,
      createBookmarkAction,
      deleteBookmarksActions,
    },
    dispatch,
  ),
});

export default connect(mapStateToprops, mapDispatchToProps)(ReadSingleArticle);
