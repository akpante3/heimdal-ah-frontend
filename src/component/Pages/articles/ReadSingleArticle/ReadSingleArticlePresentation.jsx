/* eslint-disable import/no-named-as-default */
/* eslint-disable arrow-body-style */
/* eslint-disable no-lone-blocks */
/* eslint-disable no-unused-expressions */
/* eslint-disable no-nested-ternary */
/* eslint-disable no-return-assign */
import React, { Component, Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import TimeAgo from 'javascript-time-ago';
import moment from 'moment';
import { connect } from 'react-redux';
import english from 'javascript-time-ago/locale/en';
import Glow from '../../../ui/Buttons/glow/Glow';
import './ReadSingleArticle.scss';
import ArticlesRating from '../../../ui/Rating/ArticlesRating';
import ReportArticle from '../../../ui/Forms/ReportArticle/Index';
import ModalButton from '../../../ui/Modal/Index';

TimeAgo.addLocale(english);
const timeAgo = new TimeAgo('en-US');

/**
 * @description - ReadSingleArticlePresentation renders Reading Single article page
 */
class ReadSingleArticlePresentation extends Component {
  constructor(props) {
    super(props);
    this.state = {};

    this.editButton = React.createRef();
    this.followButton = React.createRef();
  }

  /**
   * @description - render single article page
   */
  render() {
    const { author = {}, auth } = this.props;
    const username = `${this.props.username.substring(0, 9)}...`;

    return (
      <Fragment>
        <Redirect to={`/${this.props.username}/articles/${this.props.slug}`} />
        <div className="single-article-wrapper">
          <div className="article-header">
            <h1>{this.props.title}</h1>
          </div>
          <div className="bookmark-row">
            <div />
            <div>
              <img className="bookmark-logo" src="/src/images/bookmark.svg" alt="bookmark-logo" />
            </div>
          </div>
          <div className="body-section">
            <div className="article-section">
              {this.props.body}
              <div className="bottom-details">
                <div>
                  {this.props.articleId && <ArticlesRating articleId={this.props.articleId} />}
                </div>
                <div className='ratings' />
                <div className='glow'>
                  <Glow active={this.props.active} handleGlow={this.props.handleGlow} />
                  <span className='likesCount'>{ this.props.likesCount }</span>
                </div>
              </div>
              <div className="line" />
              <div className="col-md-12">
                {auth.isAuthenticated
                  && (!author ? (
                    <ModalButton label="Report Article" Class="btn-danger p-2 ph-25">
                      <ReportArticle articleId={this.props.articleId} />
                    </ModalButton>
                  ) : null)}
              </div>
            </div>
            <div className="user-section">
              <div className="user-profile">
                <img className="user-image" src={this.props.userImage} alt="user" />
                <br />
                <br />
                <a href={`/${this.props.username}`} className="user-link">
                  <h2>{username}</h2>
                </a>
                <br />
                <h4>{timeAgo.format(moment(this.props.time).valueOf())}</h4>
                <br />

                {auth.isAuthenticated
                  && (author ? (
                    <Link
                      to={`/update-articles/update?id=${this.props.articleId}`}
                      className="btn btn-secondary"
                    >
                      edit
                    </Link>
                  ) : (
                    <button className="btn follow-btn" type="submit">
                      follow
                    </button>
                  ))}
                <br />
                <br />
                <div className="social-media">
                  <i className="fab fa-4x fa-facebook-square facebook" />
                  <i className="fab fa-4x fa-twitter-square twitter" />
                </div>
              </div>
              <div>
                <hr />
                <h5>tags</h5>
                <div className="tag-buttons">
                  <div className="radio">
                    <input type="radio" name="optradio" checked />
                  </div>
                  <div className="radio">
                    <input type="radio" name="optradio" />
                  </div>
                  <div className="radio disabled">
                    <input type="radio" name="optradio" disabled />
                  </div>
                  <div className="radio">
                    <input type="radio" name="optradio" checked />
                  </div>
                  <div className="radio">
                    <input type="radio" name="optradio" />
                  </div>
                  <div className="radio disabled">
                    <input type="radio" name="optradio" disabled />
                  </div>
                  <div className="radio">
                    <input type="radio" name="optradio" checked />
                  </div>
                  <div className="radio">
                    <input type="radio" name="optradio" />
                  </div>
                  <div className="radio disabled">
                    <input type="radio" name="optradio" disabled />
                  </div>
                </div>
              </div>
              <div>
                <hr />
                <h5>Recent articles</h5>
              </div>
            </div>
            <div />
            <br />
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps)(ReadSingleArticlePresentation);
