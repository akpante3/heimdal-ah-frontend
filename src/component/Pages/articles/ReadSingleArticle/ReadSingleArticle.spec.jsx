/* eslint-disable no-undef */
import React from 'react';
import { ReadSingleArticle } from './ReadSingleArticle';
import ReadSingleArticlePresentation from './ReadSingleArticlePresentation';
import ReadSingleArticleError from './ReadSingleArticleError';


describe('Single Article Component Test Suite', () => {
  describe('<ReadSingleArticle />', () => {
    it('should render the ReadSingleArticle component', () => {
      const match = {
        params: {
          slug: 'hello',
        },
      };
      const actions = {
        getArticleById: () => {},
      };
      const singleArticle = {
        id: 2,
      };
      // eslint-disable-next-line max-len
      const wrapper = shallow(<ReadSingleArticle match={match} actions={actions} singleArticle={singleArticle} />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('<ReadSingleArticlePresentation />', () => {
    it('should render the ReadSingleArticlePresentation component', () => {
      const wrapper = shallow(<ReadSingleArticlePresentation />);
      expect(wrapper.length).toBe(1);
    });
  });

  describe('<ReadSingleArticlePresentationError />', () => {
    it('should render the ReadSingleArticleError component', () => {
      const wrapper = shallow(<ReadSingleArticleError />);
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
