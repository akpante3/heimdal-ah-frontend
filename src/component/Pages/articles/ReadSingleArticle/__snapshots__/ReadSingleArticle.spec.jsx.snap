// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Article Component Test Suite <ReadSingleArticle /> should render the ReadSingleArticle component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReadSingleArticle
    actions={
      Object {
        "getArticleById": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "slug": "hello",
        },
      }
    }
    singleArticle={
      Object {
        "id": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <React.Fragment>
        <Connect(ReadSingleArticlePresentation)
          active={false}
          articleId=""
          author={false}
          body=""
          handleGlow={[Function]}
          likesCount={0}
          slug="hello"
          title=""
          username="@username"
        />
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Connect(ReadSingleArticlePresentation)
          active={false}
          articleId=""
          author={false}
          body=""
          handleGlow={[Function]}
          likesCount={0}
          slug="hello"
          title=""
          username="@username"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "articleId": "",
          "author": false,
          "body": "",
          "handleGlow": [Function],
          "likesCount": 0,
          "slug": "hello",
          "time": undefined,
          "title": "",
          "userImage": undefined,
          "username": "@username",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <React.Fragment>
          <Connect(ReadSingleArticlePresentation)
            active={false}
            articleId=""
            author={false}
            body=""
            handleGlow={[Function]}
            likesCount={0}
            slug="hello"
            title=""
            username="@username"
          />
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(ReadSingleArticlePresentation)
            active={false}
            articleId=""
            author={false}
            body=""
            handleGlow={[Function]}
            likesCount={0}
            slug="hello"
            title=""
            username="@username"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "articleId": "",
            "author": false,
            "body": "",
            "handleGlow": [Function],
            "likesCount": 0,
            "slug": "hello",
            "time": undefined,
            "title": "",
            "userImage": undefined,
            "username": "@username",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Single Article Component Test Suite <ReadSingleArticlePresentation /> should render the ReadSingleArticlePresentation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ReadSingleArticlePresentation)
    username="@username"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Single Article Component Test Suite <ReadSingleArticlePresentationError /> should render the ReadSingleArticleError component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReadSingleArticleError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Sorry This Article
          <br />
          Sorry This Article Does Not Exist
        </h3>,
        <br />,
        <a
          to="/"
        >
          <div>
            ..go back Home....
          </div>
        </a>,
      ],
      "className": "container ReadSingleArticleError",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sorry This Article",
            <br />,
            "Sorry This Article Does Not Exist",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sorry This Article",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Sorry This Article Does Not Exist",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            ..go back Home....
          </div>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "..go back Home....",
          },
          "ref": null,
          "rendered": "..go back Home....",
          "type": "div",
        },
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Sorry This Article
            <br />
            Sorry This Article Does Not Exist
          </h3>,
          <br />,
          <a
            to="/"
          >
            <div>
              ..go back Home....
            </div>
          </a>,
        ],
        "className": "container ReadSingleArticleError",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sorry This Article",
              <br />,
              "Sorry This Article Does Not Exist",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sorry This Article",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Sorry This Article Does Not Exist",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              ..go back Home....
            </div>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "..go back Home....",
            },
            "ref": null,
            "rendered": "..go back Home....",
            "type": "div",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
